pkg/build/api/validation/validation.go

```diff
@@ -683,7 +683,12 @@ func ValidateStrategyEnv(vars []kapi.EnvVar, fldPath *field.Path) field.ErrorLis
 			allErrs = append(allErrs, field.Invalid(idxPath.Child("name"), ev.Name, cIdentifierErrorMsg))
 		}
 		if ev.ValueFrom != nil {
-			allErrs = append(allErrs, field.Invalid(idxPath.Child("valueFrom"), ev.ValueFrom, "valueFrom is not supported in build strategy environment variables"))
+			if ev.Value != "" {
+				allErrs = append(allErrs, field.Duplicate(idxPath.Child("value"), fmt.Sprintf("cannot use 'value' and 'valueFrom' together for %v", ev.Name)))
```

Comment:
AFAICS Kubernetes allows the user to specify value and valueFrom.  If value is set, it takes precedence.  I can see advantages in behaving in the same way, and I can see also advantages in being more restrictive. @bparees do you have an opinion on this?